bach/bach/series/series.py

```diff
@@ -673,6 +673,28 @@ def to_numpy(self) -> numpy.ndarray:
         """
         return self.to_pandas().to_numpy()
 
+    def reset_index(
+        self,
+        level: Optional[Union[str, Sequence[str]]] = None,
+        drop: bool = False,
+    ) -> DataFrameOrSeries:
+        """
+        Drops the current index.
+
+        :param level: Removes given levels from index. Removes all levels by default
+        :param drop: if False, the dropped index is added to the data columns of the DataFrame. If True it
+            is removed.
+        :returns: DataFrame with the index dropped.
+        """
+        series = self.copy()
+
+        result = series.to_frame().reset_index(level, drop)
```

Comment:
`self.to_frame()` already copies the series, so no need to create another copy:)
```suggestion
        result = self.to_frame().reset_index(level, drop)
```